registration/registration-services/src/main/java/io/mosip/registration/update/SoftwareUpdateHandler.java

```diff
@@ -548,96 +551,97 @@ public Timestamp getLatestVersionReleaseTimestamp() {
+	 * local
 	 * </p>
 	 * <p>
 	 * If there is any error occurs during the update,then the rollback query will
 	 * run from the sql file
 	 * </p>
 	 * 
-	 * @param actualLatestVersion
-	 *            latest version
 	 * @param previousVersion
 	 *            previous version
+	 * @param versionMappings 
 	 * @return response of sql execution
 	 * @throws IOException
 	 */
-	public ResponseDTO executeSqlFile(String actualLatestVersion, String previousVersion) throws IOException {
-
-		LOGGER.info("DB-Script files execution started from previous version : {} , To Current Version : {}",previousVersion, currentVersion);
-
-		String newVersion = actualLatestVersion.split("-")[0];
-		previousVersion = previousVersion.split("-")[0];
+	public ResponseDTO executeSqlFile(String previousVersion, Map<String, VersionMappings> versionMappings) throws IOException {
+		LOGGER.info("DB-Script files execution started from previous version : {} , To Current Version : {}", previousVersion, currentVersion);
 		
-
 		ResponseDTO responseDTO = new ResponseDTO();
+		
+		/*
+		 * Here, we are removing the entries from version-mappings map, for which, the
+		 * releaseOrder is less than or equal to the previous version, because, we need
+		 * to execute the upgrade scripts only for the versions released after the
+		 * previous version.
+		 */
+		if (versionMappings.containsKey(previousVersion)) {
+			Integer previousVersionReleaseOrder = versionMappings.get(previousVersion).getReleaseOrder();
+			versionMappings.entrySet().removeIf(matches -> matches.getValue().getReleaseOrder() <= previousVersionReleaseOrder);
+		}
+		
+		for (Entry<String, VersionMappings> entry : versionMappings.entrySet()) {
+			try {
+				executeSQL(entry.getValue().getDbVersion());
+				previousVersion = entry.getKey();
+				// Update global param with current version
+				globalParamService.update(RegistrationConstants.SERVICES_VERSION_KEY, entry.getKey());
+			} catch (RegBaseCheckedException exception) {
+				// Prepare Error Response
+				setErrorResponse(responseDTO, RegistrationConstants.SQL_EXECUTION_FAILURE, null);
+				// Replace with backup
+				rollback(responseDTO, previousVersion);
+				return responseDTO;
+			}
+		}
+		
+		setSuccessResponse(responseDTO, RegistrationConstants.SQL_EXECUTION_SUCCESS, null);
+
+		LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
+				"DB-Script files execution completed");
 
-		// execute sql file
+		return responseDTO;
+	}
 
+	private void executeSQL(String dbVersion) throws RegBaseCheckedException {
 		try {
-
 			LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
-					"Checking Started : " + newVersion + SLASH + exectionSqlFile);
+					"Checking Started : " + dbVersion + SLASH + exectionSqlFile);
 
-			execute(SQL + SLASH + newVersion + SLASH + exectionSqlFile);
+			execute(SQL + SLASH + dbVersion + SLASH + exectionSqlFile);
```

Comment:
After the successful execution of the script, we need to take a backup of migrated DB

Reply:
Yes, added logic to take backup after every successful execution.